content/en/blog/2024/policing-a-sandbox.md

```diff
@@ -0,0 +1,296 @@
+
+Hence a pure software approach to establish trust in Init-Data is not desirable. We want to move the trust boundary back to the TEE and link Init-Data measurements to the TEE’s hardware evidence. There are generally 2 options to establish such a link, which one of those is chosen depends on the capabilities of the TEE:
+
+### Host-Data
+
+Host-Data is a field in a TEE’s evidence that is passed into a confidential Guest from its Host verbatim. It’s not secret, but its integrity is guaranteed, as its part of the TEE-signed evidence body. We are generalising the term Host-Data from SEV-SNP here, a similar concept exists in other TEEs with different names. Host-Data can hold a limited amount of bytes, typically in the 32 - 64 byte range. This is enough to hold a hash of the Init-Data, calculated at the launch of the Guest. This hash can be used to verify the integrity of the Init-Data in the guest, by comparing the measurement (hash) of the Init-Data in the guest with the host-provided hash in the Host-Data field. If the hashes match, the Init-Data is considered to be intact.
+
+Example: Producing a SHA256 digest of the Init-Data file
+
+```bash
+openssl dgst -sha256 --binary init-data.toml | xxd -p -c32
+bdc9a7390bb371258fb7fb8be5a8de5ced6a07dd077d1ce04ec26e06eaf68f60
+```
+
+### Runtime Measurements
+
+Instead of seeding the Init-Data hash into a Host-Data field at launch, we can also extend the TEE evidence with a runtime measurement of the Init-Data directly, if the TEE allows for it. This measurement is then a part of the TEE’s evidence and can be verified as part of the TEE’s remote attestation process.
+
+Example: Extending an empty SHA256 runtime measurement register with the digest of an Init-Data file 
+
+```bash
+dd if=/dev/zero of=zeroes bs=32 count=1
+openssl dgst -sha256 --binary init-data.toml > init-data.digest
+openssl dgst -sha256 --binary <(cat zeroes init-data.digest) | xxd -p -c32
+7aaf19294adabd752bf095e1f076baed85d4b088fa990cb575ad0f3e0569f292
+```
+
+## Glueing Things Together
+
+Finally, in practice a workflow would look like the steps depicted below. Note that the concrete implementation of the individual steps might vary in future revisions of CoCo, so this is not to be taken as a reference, but merely to illustrate the concept. There are practical considerations, like limitations in the size of a Pod annotation, or how Init-Data can be provisioned into a guest that might alter details of the workflow in the future.
```

Comment:
Fair enough to mention it can change, but it'd be nice to have at least a hint to when this was written (as in, a specific version of the project).